module/VuFind/src/VuFind/I18n/Translator/TranslatorAwareTrait.php

```diff
@@ -114,7 +117,12 @@ public function translate($target, $tokens = [], $default = null)
                 return $translated;
             }
             // Override $domain/$str using getDisplayString() before proceeding:
-            list($domain, $str) = $this->extractTextDomain($str->getDisplayString());
+            $str = $str->getDisplayString();
```

Comment:
This is potentially confusing -- translatable strings inside translatable strings? Might be worth adding a comment or two to explain why this check is necessary.

Reply:
I added a comment explaining this a bit. This is indeed convoluted, but there are two reasons for this:

1.) HierarchicalFacetListener can process the facet values in the record and make them translatable without losing the original value, while having no idea whether the translated version is to be displayed and how. And still provide a default translation based on translating the facet levels independently.

2.) Allowing display string to be a TranslatableString makes it possible to have more than one potential translation key while maintaining the order of them. This is needed so that a facet value can be translated using the full facet value (e.g. "1/Book/Book Chapter/") or the default display value (e.g. "Book/Book Chapter"). The first one is precise and has always been supported, so it's important to keep it for back-compatibility. The latter one allows adding translation based on what you see on screen.

Reply:
Makes sense; thanks!